kore/src/Kore/Builtin/List.hs

```diff
@@ -512,7 +519,8 @@ unifyEquals
         builtin1
         builtin2
         frame2
-      | Seq.length prefix2 > Seq.length list1 =
+      | Seq.length prefix2 > Seq.length list1 = do
+        explainWithCannotUnifyDifferentLength
```

Comment:
You could define a helper that gives the explanation and also returns `\bottom`:
```.hs
cannotUnifyDifferentLength = do
    Monad.Unify.explainBottom
        (Pretty.hsep
            [ "Cannot unify lists of different length."
            ]
        )
        first
        second
    return (ExpandedPattern.bottom, SimplificationProof)
```